pyramid_hypernova/request.py

```diff
@@ -73,7 +76,10 @@ def json(self):
                 self.response.raise_for_status()
                 json = self.response.json()
             except (HTTPError, ConnectionError) as e:
-                raise HypernovaQueryError(e)
+                error_data = None
+                if hasattr(self, 'response'):
+                    error_data = self.response.json().get('error', None)
```

Comment:
a comment here explaining when and why we're doing this would be good

personally i'm curious if `.response` is always available on a certain class of error or if it's sometimes missing? if it's always available on `HTTPError`, for example, then any reason we're not splitting the `except` to catch different error types and handle them separately?

Reply:
There's an existing test that throws an error before self.response = requests.post can even run. This implementation was to satisfy that test, though in hindsight it might be better to just rework/remove that test. 🤔